src/sentry/static/sentry/app/components/stackedBarChart.tsx

```diff
@@ -327,38 +342,71 @@ class StackedBarChart extends React.Component<Props, State> {
 
   renderChartColumn(
     point: Point,
+    secondaryPoint: Point | null,
     maxval: number,
     pointWidth: number,
     index: number,
     _totalPoints: number
   ): React.ReactNode {
-    const totalY = point.y.reduce((a, b) => a + b);
+    const {showSecondaryPoints} = this.props;
+
+    const totalY = (secondaryPoint ? secondaryPoint : point).y.reduce((a, b) => a + b);
     const totalPct = totalY / maxval;
     // we leave a little extra space for bars with min-heights.
     const maxPercentage = 99;
 
     let prevPct = 0;
+    let prevPct2 = 0;
     const pts = point.y.map((y, i) => {
       const pct = Math.max(
         totalY && formatFloat((y / totalY) * totalPct * maxPercentage, 2),
         this.getMinHeight(i)
       );
 
+      const y2 = secondaryPoint && secondaryPoint.y[i];
+      const pct2 =
+        y2 &&
+        Math.max(
+          totalY && formatFloat((y2 / totalY) * totalPct * maxPercentage, 2),
+          this.getMinHeight(i)
+        );
+
       const pt = (
-        <rect
-          key={i}
-          x={index * pointWidth + '%'}
-          y={100.0 - pct - prevPct + '%'}
-          width={pointWidth - this.props.gap + '%'}
-          data-test-id="chart-column"
-          height={pct + '%'}
-          fill={this.state.series[i].color}
-          className={classNames(this.props.barClasses[i], 'barchart-rect')}
-        >
-          {y}
-        </rect>
+        <React.Fragment key={i}>
+          <rect
```

Comment:
Part of me wonder if we would be better off replacing this hand rolled chart with an echarts based implementation instead in the future 🤔.

Reply:
I thought similarly as well also, tried to use `y` as an array with two values zipped together but then I'd still have to manipulate y-array values on hover, and change the color implementation and it would be a similar amount of work fixing the other uses of stackedbarchat

Reply:
Maybe converting this to echarts will allow that robustness without having to fiddle with flags and data lengths, I can add that to post internal-release discussion if it's not a blocker for that (sentry-internal release)

Reply:
Changing the charts over is totally something that can and should happen separately.

Reply:
You can follow the progress here (WOR-238)[https://getsentry.atlassian.net/browse/WOR-238]